src/configurationProvider.ts

```diff
@@ -209,6 +210,7 @@ export class JavaDebugConfigurationProvider implements vscode.DebugConfiguration
                 config.vmArgs = this.concatArgs(config.vmArgs);
             }
 
+            await populateShortenCommandLineField(config);
```

Comment:
Does this only apply to the launch scenario? If so, it should come together with the launch logic section.